### PR TITLE
Issue 28656 inline translation rest api

### LINKED ISSUE #28656: Enabled Translation inline affect to result of REST APIs.
### Preconditions (*)

1. Magento 2.3.4

### Steps to reproduce (*)

1. Go to admin
2. Stores > Settings > Configuration
3. In the left panel, expand Advanced and choose Developer
4. Expand selector the Translate Inline section
5. Set Enabled for Storefront to Yes
6. Click on save config
7. Try to login to admin as a user using API-
POST -  {Baseurl}/rest/V1/integration/admin/token
Body-{
"username": "",
"password": ""
}

### Expected result (*)

1. Translation inline should be effected to frontend side only.
2. Message from REST API should work well while translation inline is enabled.

### Actual result (*)

1.  Result from rest API:
`{
    "message": "<span data-translate=\"[&#x7B;&quot;shown&quot;&#x3A;&quot;Invalid&#x20;login&#x20;or&#x20;password.&quot;,&quot;translated&quot;&#x3A;&quot;Invalid&#x20;login&#x20;or&#x20;password.&quot;,&quot;original&quot;&#x3A;&quot;Invalid&#x20;login&#x20;or&#x20;password.&quot;,&quot;location&quot;&#x3A;&quot;Text&quot;,&quot;scope&quot;&#x3A;&quot;Invalid&#x20;login&#x20;or&#x20;password.&#x7D;&#x7D;&#x7B;&#x7B;theme&quot;&#x7D;]\">Invalid login or password.</span>"
}`

### Additional information
- more detailed steps to reproduce with screens https://github.com/magento/magento2/issues/28656#issuecomment-644551601

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._

### PR DESCRIPTION
**Pull request by zaximus84**
### Description (*)
When inline translation is enabled for the storefront, it should be limited to the frontend where HTML is rendered. However, making an API call to /rest/V1/integration/admin/token with bad credentials while inline translation is enabled results in inline translation markup around the error message in the API response.

Since inline translation markup is only useful in the adminhtml and frontend areas, this PR adds an area check in addition to the existing configuration checks before rendering inline translation markup.

### Fixed Issues (if relevant)
1. Fixes magento/magento2#28656

### Manual testing scenarios (*)
1. Go to admin
2. Stores > Settings > Configuration
3. In the left panel, expand Advanced and choose Developer
4. Expand selector the Translate Inline section
5. Set Enabled for Storefront to Yes
6. Click on save config
7. Try to login to admin as a user using API-
    POST - {Baseurl}/rest/V1/integration/admin/token
    Body - {"username": "","password": ""}
8. Confirm the response does not contain inline translation markup (a <span>)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: lib/internal/Magento/Framework/Translate/Inline.php [hunk 1]
```diff
@@ -249,4 +266,21 @@ protected function stripInlineTranslations(&$body)
         }
         return $this;
     }
+
+    /**
+     * Indicates whether the current area is valid for inline translation
+     *
+     * @return bool
+     */
+    private function isAreaAllowed()
```
Comment (swnsma) [inline lib/internal/Magento/Framework/Translate/Inline.php]:
```suggestion
    private function isAreaAllowed(): bool
```

File: lib/internal/Magento/Framework/Translate/Inline.php [hunk 2]
```diff
@@ -89,6 +103,7 @@ public function __construct(
         \Magento\Framework\Translate\Inline\ConfigInterface $config,
         \Magento\Framework\Translate\Inline\ParserInterface $parser,
         \Magento\Framework\Translate\Inline\StateInterface $state,
+        \Magento\Framework\App\State $appState,
```
Comment (swnsma) [inline lib/internal/Magento/Framework/Translate/Inline.php]:
```suggestion
        \Magento\Framework\App\State $appState = null,
```

File: lib/internal/Magento/Framework/Translate/Inline.php [hunk 3]
```diff
@@ -99,6 +114,7 @@ public function __construct(
         $this->config = $config;
         $this->parser = $parser;
         $this->state = $state;
+        $this->appState = $appState;
```
Comment (swnsma) [inline lib/internal/Magento/Framework/Translate/Inline.php]:
```suggestion
        $this->appState = $appState ?: \Magento\Framework\App\ObjectManager::getInstance()
               ->(\Magento\Framework\App\State::class);
```

**Review by swnsma — CHANGES_REQUESTED** [review]
Hi @zaximus84!
Please check my suggestions.
Also, please check failed tests.

**Comment by swnsma** [comment]
@magento run all tests

**Comment by swnsma** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
Hi @zaximus84, can You please fix failed tests and look through requested changes?

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by swnsma** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by swnsma — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @swnsma, thank you for the review.
[ENGCOM-7829](https://jira.corp.magento.com/browse/ENGCOM-7829) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Go to Admin-> Stores -> Configuration -> Advanced -> Developer -> Translate Inline;
2. Set **Enabled for Storefront** to **Yes** and Save;
3. Clean cache;
4. Go to Storefront -> Sign In

Actual Result: :heavy_check_mark:  Inline translate for Storefront is enabled

Screenshot from 2020-07-14 12-02-12

5. Try to login to admin as a user using API-
**POST** - `{Baseurl}/rest/V1/integration/admin/token`
Body - {"username": "","password": ""}

**Before:** :heavy_multiplication_x: Result from rest API:



**After:** :heavy_check_mark: Inline translation only affects frontend side and the response does not contain inline translation markup

```
{
  "message": "The account sign-in was incorrect or your account is disabled temporarily. Please wait and try again later.",
}
```

**Pull request merged**